GiniSwitchSDK/Example/GiniTariffSDK/ExtractionsViewController.swift

```diff
@@ -23,7 +23,67 @@ class ExtractionsViewController: UIViewController {
+                self?.extractionsCollection?.customerAddress?.value?.country = changedValue
+            }))
+            allExtractions.append((name: "Customer postal code", value: extractionsCollection?.customerAddress?.value?.postalCode, changeBlock: { [weak self] (changedValue) in
+                self?.extractionsCollection?.customerAddress?.value?.postalCode = changedValue
+            }))
+            allExtractions.append((name: "Energy meter number", value: extractionsCollection?.energyMeterNumber?.value?.valueString, changeBlock: { [weak self] (changedValue) in
+                self?.extractionsCollection?.energyMeterNumber?.value = changedValue
+            }))
+            allExtractions.append((name: "Comsumption", value: extractionsCollection?.consumption?.value?.valueString, changeBlock: { [weak self] (changedValue) in
+                if let numberValue = Double(changedValue) {
+                    self?.extractionsCollection?.consumption?.value?.value = numberValue
+                }
+            }))
+            allExtractions.append((name: "Consumption duration", value: extractionsCollection?.consumptionDuration?.value?.valueString, changeBlock: { [weak self] (changedValue) in
+                if let numberValue = Double(changedValue) {
+                    self?.extractionsCollection?.consumptionDuration?.value?.value = numberValue
+                }
+            }))
+            allExtractions.append((name: "Paid amount", value: extractionsCollection?.paidAmount?.value?.valueString, changeBlock: { [weak self] (changedValue) in
+                if let numberValue = Double(changedValue) {
+                    self?.extractionsCollection?.paidAmount?.value?.value = numberValue
+                }
+            }))
+            allExtractions.append((name: "Amount to pay", value: extractionsCollection?.amountToPay?.value?.valueString, changeBlock: { [weak self] (changedValue) in
+                if let numberValue = Double(changedValue) {
+                    self?.extractionsCollection?.amountToPay?.value?.value = numberValue
+                }
+            }))
+            allExtractions.append((name: "Billing amount", value: extractionsCollection?.billingAmount?.value?.valueString, changeBlock: { [weak self] (changedValue) in
+                if let numberValue = Double(changedValue) {
+                    self?.extractionsCollection?.billingAmount?.value?.value = numberValue
+                }
+            }))
+            allExtractions.append((name: "Document date", value: extractionsCollection?.documentDate?.value?.valueString, changeBlock: { [weak self] (changedValue) in
+                self?.extractionsCollection?.documentDate?.value = changedValue
+            }))
+            extractions = allExtractions.filter({ $0.name != nil && $0.value != nil}).map({ (element) in
```

Comment:
Why filtering and mapping? You can use a `flatMap` to do the same, like so:
```
allExtractions.flatMap {
    if let name = $0.name, let value = $0.value {
        return (name: name, value: value, changeBlock:$0.changeBlock)
    }
    return nil
}
```

It should have also better performance, although it is an example app.